compiler/jsgen.nim

```diff
@@ -1655,15 +1655,15 @@ proc genVarInit(p: PProc, v: PSym, n: PNode) =
     varCode: string
     varName = mangleName(p.module, v)
     useReloadingGuard = sfGlobal in v.flags and p.config.hcrOn
-
+  let jsVarDecl = isDefined(p.config, "nimJsVar")
   if v.constraint.isNil:
     if useReloadingGuard:
-      lineF(p, "var $1;$n", varName)
+      if jsVarDecl: lineF(p, "var $1;$n", varName) else: lineF(p, "let $1;$n", varName)
```

Comment:
here + elsewhere, please make it more DRY:
introduce this once: `let jsdecl = if jsVarDecl: "var" else: "let"`
then use it everywhere relevantm eg:

lineF(p, "$1 $2;$n", [jsdecl, varName])